### ISSUE TITLE
packaging: Release rpm for Fedora/RHEL/...

### ISSUE THREAD

**Issue by TS-CUBED** [quarto-dev/quarto-cli#915]
Is there a chance to get an rpm package as well?

I just converted the deb using `alien` and the resulting rpm works nicely (using the `--scripts` option). Maybe include in a copr?

Edit: tested on Fedora 36, had already installed pandoc and texlive.

**Comment by bhogan-mitre** [comment]
Is there any update on this now that we're past a 1.0 release?  Also are there implied system dependencies for compiler versions?  So far my closest attempt at an install has been to convert deb to rpm as described above, but I still ran into an issue with GLIBC versions.

```
Error: Package: Quarto-1.0.38-2.x86_64 (/Quarto-1.0.38-2.x86_64)
           Requires: libc.so.6(GLIBC_2.18)(64bit)
```

**Comment by jjallaire** [comment]
@msarahan Has created a build of Deno that uses the older version of glibc. @msarahan What is the link to that binary?

@dragonstyle We may want to start creating a tar.gz for RHEL that is the same as what we now create for Debian/Ubuntu but has the custom built Deno binary.

**Comment by msarahan** [comment]
The conda-forge package for deno version 1.22.0 (matching quarto 1.1.x) is available at https://anaconda.org/conda-forge/deno/1.22.0/download/linux-64/deno-1.22.0-hf47965d_0.tar.bz2

You don't need conda to use that package. Just download it and extract it somewhere. You can either set the `QUARTO_DENO` environment variable to point to the binary in there, or you can replace the deno binary in the quarto archive with the one from the conda package.

Please tell us how this works for you. We are finding an issue with the combination of conda-built packages where sass is not executing correctly. If you do see any odd behavior, please tag me on a comment here or in a new issue.

**Comment by jjallaire** [comment]
@dragonstyle I tagged this for v1.1 (which should just entail adding a github action that creates a RHEL tar.gz that swaps our deno binary for the one here (https://anaconda.org/conda-forge/deno/1.22.0/download/linux-64/deno-1.22.0-hf47965d_0.tar.bz2). I believe that RStudio will also be very interested in taking this once it's availble.

Note that we should _only_ do this on v1.1 for right now because we are about to update to Deno 1.25.1 on v1.2 and we'll want to wait for a conda version of that.

**Comment by msarahan** [comment]
I will keep an eye out for 1.25.1 and get it built when it comes out. So far I only see 1.25.0.

**Comment by jjallaire** [comment]
I believe they are planning on releasing 1.25.1 later on today.

**Comment by dragonstyle** [comment]
This is completed on the 1.1 branch. Moving to 1.2 to track updating to 1.25...

**Comment by benformosa** [comment]
Was the outcome of this issue releasing the `linux-rhel7-amd64.tar.gz` tarball?
Is there any plan for an RPM?

**Comment by benformosa** [comment]
I have had success building an RPM with [fpm](https://github.com/jordansissel/fpm):

```bash
VERSION="$(curl -s https://api.github.com/repos/quarto-dev/quarto-cli/releases/latest | python3 -c 'import json,sys;print(json.load(sys.stdi
n)["name"].lstrip("v"))')"

URL="https://github.com/quarto-dev/quarto-cli/releases/download/v${VERSION}/"
TARBALL="quarto-${VERSION}-linux-rhel7-amd64.tar.gz"
RELEASE=1

SCRIPT_POST='ln -fs /opt/quarto/bin/quarto /usr/local/bin/quarto
ln -fs /opt/quarto/share/man/quarto-man.man /usr/share/man/man1/quarto.1'
SCRIPT_POSTUN='rm -f /usr/local/bin/quarto /usr/share/man/man1/quarto.1'

curl --location --continue-at - --output "${TARBALL}" "${URL}${TARBALL}"
tar -x --skip-old-files -f "${TARBALL}"

fpm \
    --verbose \
    --rpm-digest sha256 \
    --input-type dir \
    --output-type rpm \
    --package "quarto-${VERSION}-${RELEASE}.x86_64.rpm" \
    --name quarto \
    --license GPL-2.0-or-later \
    --version "${VERSION}" \
    --iteration "${RELEASE}" \
    --architecture x86_64 \
    --description "Quarto is an academic, scientific, and technical publishing system built on Pandoc." \
    --url https://github.com/quarto-dev/quarto-cli \
    --after-install <(echo "${SCRIPT_POST}") \
    --after-remove <(echo "${SCRIPT_POSTUN}") \
    "quarto-${VERSION}/bin=/opt/quarto" \
    "quarto-${VERSION}/share=/opt/quarto"
```

**Comment by erentar** [comment]
I would like to second this, official rpm packages would be a lovely and needed addition

**Comment by mirenradia** [comment]
Can this issue be re-opened since there isn't an official RPM package available yet and it would be great to have one, @dragonstyle?

For now, I have resorted to using alien to convert the deb package as per the OP.
